internal/services/keyvault/key_vault_key_resource.go

```diff
@@ -62,7 +63,7 @@ func resourceKeyVaultKey() *pluginsdk.Resource {
 				Type:         pluginsdk.TypeString,
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: keyVaultValidate.VaultID,
+				ValidateFunc: validation.Any(keyVaultValidate.VaultID, keyVaultValidate.ManagedHSMID),
```

Comment:
Unfortunately as discussed in #20150 - we can't shoe-horn this into the existing resource since the two API's are already diverging, see: https://github.com/Azure/azure-sdk-for-go/pull/20296/files/a5b8068678e35dfe5e4941294897335e88cf3b8e#diff-14f54f17b4ccdfeb282de9756ca15ec92e75d473d8b583a1bc54439d62ddc6b7R3

Instead as mentioned in #20150, we'll need to introduce a new resource specifically for a Managed HSM Key

Reply:
only the RBAC API for Managed HSM is diverged from keyvault if my understanding right? according to the [#uses-same-api-and-management-interfaces-as-key-vault](https://learn.microsoft.com/en-us/azure/key-vault/managed-hsm/overview#uses-same-api-and-management-interfaces-as-key-vault), the key management API is the same.

it makes sense the use a separate resource type (but with the same under client instance) to manage keys for Managed HSM.

Reply:
thanks for your explanation! i wonder if I should just register a new resource type with the existing `key_vault_key_source.go`, or register a new resource with copy-then-modify the key_vault_key resource.

both implements should not introduce breaking change in the future even if the API diverged eventually. What do you think about it?

Reply:
Copying the existing resource but updating it to be specific to Managed HSM should be fine - although I'd suggest doing that in a separate PR since this one's also adding the role definition/assignment resources?